### ISSUE TITLE
page not opening in iOS.

### ISSUE THREAD

**Issue by dommermuth** [myflashlab/webView-ANE#41]
Hello,
I have used your FB (Flash Builder) example here at Github and though it works nicely for Android it fails for iOS.  

In order for your example to run it required some simple modifications, but the code related to the ANE is unchanged with the exception of un-commenting the _ex.openWebViewURL line and commenting out the line before it.

I also edited the xml to point to AIR SDK 19.

The ANE I'm using came packaged in a folder labeled richWebview_V4.9.

In the screen shot (iOS 9,1) (attached) it shows that the demo button was clicked a couple times.  It is attempting to load google.com, but the Google page never shows. However, as you can see, the onPageFinished event fires.

I have the exact same problem in another app I'm running.

Can you please test this?  Please let me know what other information I can provide.

Thank you.
![img_0069](https://cloud.githubusercontent.com/assets/3721352/11746004/b6499464-9fe9-11e5-8ce6-4eb38d6f6df8.jpg)

**Comment by myflashlab** [comment]
@dommermuth Please notice to always check the FlashDevelop sample files as FD is the main IDE we use. (we are organizing to keep FlashBuilder and FlashCC updated too soon)

please check the following:
make sure your Air manifest has all the settings for RichWebview by checking https://github.com/myflashlab/webView-ANE/blob/master/FD/application.xml

and to make sure you are correctly initializing the extension, please check this sample file https://github.com/myflashlab/webView-ANE/blob/master/FD/src/Demo.as

some thoughts: you screenshot shows the page is loading but if you can't see it, I think you should check the webview location while opening the page ```_ex.openWebViewURL(0, 0, stage.stageWidth, stage.stageHeight, "http://www.google.com");```

the first four parameters are x,y,width and height.

let me know how it goes.

**Comment by dommermuth** [comment]
Hello,

I used your flash develop xml and Demo.as.

The results are the same.

One thing is, I am using AIR SDK 19.  In your XML you are using 15.  

Have you tested the ANE with 19 yet?

Also, the code works on Android so it's not the webview location.  

Thank you.

**Comment by myflashlab** [comment]
Air SDK 19 should work just fine too. Please send us your .as file + Air manifest file so we can test it here locally on an iOS device. Thanks.

**Comment by dommermuth** [comment]
Files attached.

Device is gen 5 iPod running iOS 9.1
SDK 19
ANE 4.9

Thank you!

[MainFinal-app.zip](https://github.com/myflashlab/webView-ANE/files/60136/MainFinal-app.zip)

**Comment by myflashlab** [comment]
We just ran the ANE on iOS 9.1 with your files and it's just working fine! the only missing part in your manifest was ```<content>exWebView.swf</content>``` which was empty. rather than that, everything was just like the demo files and it did compile and run with no problem. do you have more information so maybe we can help you spot the problem?

maybe another thing to check is your -swf-version when compiling. please make sure it's at least 25

**Comment by dommermuth** [comment]
I don't think exWebView.swf is needed, but I put it in anyway and still it doesn't work.

Which SDK are you using?  Is you device being debugged via USB?  

I am using Flash Builder and I'm not sure there is a means to set the -swf-version.

I appreciate any help you can give.  Thank you.

**Comment by myflashlab** [comment]
we're currently running Air SDK 19. it works normal on release/debug/adHoc/distribution.

could you try to run it on other devices too to see if you're facing the same problem?

**Comment by dommermuth** [comment]
I have an old iPAD I can try.  Will set that up.  I also am installing SDK 20 right now.  

Will let you know.

**Comment by dommermuth** [comment]
Hello, unfortunately in SDK 20 I'm not able to get very far.  

It simply won't compile.  I get the following error:

Error occurred while packaging the application:
Compilation failed while executing : ld64

I have cleaned the project, restarted etc...

If I remove the ANE and related code, then it compiles.  

Works ok on Android.

**Comment by myflashlab** [comment]
I will check with SDK 20 and let you know... also did you try on any other iOS device?

**Comment by dommermuth** [comment]
I haven't gotten that far yet.  It's an iPad 1.  I don't have high hopes for it, but I'm charging it up now. 

Thanks for sticking with me on this.

**Comment by dommermuth** [comment]
The most recent version of iOS I can get on my iPad is 5.1.1.  I assume I need at least 6.1 to run the ANE.  

I'm going to revert back to an earlier SDK and try that.  Thanks again.

**Comment by dommermuth** [comment]
Compiled using AIR SDK 18.  I tested maybe 15 times.  It did succeed in revealing Google once, but I couldn't get it to happen again.

I switched back to SDK 20 and it simply won't compile.

The error again is Compilation failed while executing : ld64

**Comment by myflashlab** [comment]
@dommermuth I confirm the ```Compilation failed while executing : ld64``` error when using Air SDK 20.0.0.204 and it's happening on ALL the ANEs! so I just left a post on Adobe's forum so hopefully someone there can see and confirm the bug. https://forums.adobe.com/message/8294948#8294948

tell me if you were compiling on a windows or a Mac while trying the Air SDK 20?

rather than that, the ANE is compiling and working with no problem on SDK 19 or 18. we simply could not duplicate the problem you have. I suspect that something weired is going on in your project that is causing the problem. if it showed up once, the it means there's something going wrong in your setup. I really can't think of anything else!

**Comment by dommermuth** [comment]
Hello,  Thanks for confirming and pursuing a solution for SDK 20.  I've reverted back to 19.

I'm using windows 7.  Flash builder 4.7.

The project is essentially a duplicate of your FD example with a few minor changes to make it run in Flash Builder.  I can't imagine what else it could be.  I may try Flash Develop and see if I get better results.

Perhaps there is something unique to the iPod I'm using, but it's a pretty virgin device.  I never use it for anything but testing.  

Perhaps the debugger itself.

Any ideas welcome.

**Comment by dommermuth** [comment]
Apparently the ANE does not work with the Flash Builder debugger.

If I do two steps I can get it to run.

1. Comment out the following code:

var btn1:MySprite = createBtn("openWebView");
btn1.addEventListener(MouseEvent.CLICK, openWebViewLocal);
_list.add(btn1);
			
function openWebViewLocal(e:MouseEvent):void
{
//_ex.openWebViewLocal(0, 0, stage.stageWidth, stage.stageHeight,File.documentsDirectory.resolvePath("webviewV4.0/index.html"));
_ex.openWebViewURL(0, 0, stage.stageWidth, stage.stageHeight, "https://www.google.com");
}
			}
2.  Compile a final release IPA and install manually via iTunes.

Odd about step one.

**Comment by dommermuth** [comment]
Just a heads up.  I've had a chance to test your the example for this on a new ipad 2 mini and it simply won't launch google in debug mode.

I also had time to test it in Flash Develop on the ipad mini and ipod and the results were the same.

A release version works on iOS, but a debug version doesn't.

**Comment by dommermuth** [comment]
Sorry for the barrage of notes, but I have one more.

I can get this to consistently work if I define the RichWebView when the button is clicked.

This works:

```
var btn1:MySprite = createBtn("openWebView");
btn1.addEventListener(MouseEvent.CLICK, openWebView);
_list.add(btn1);
			
_ex = new RichWebView(stage, true, true, true,true); 
			
function openWebView(e:MouseEvent):void
{
				_ex = new RichWebView(stage, true, true, true,true);
				_ex.openWebViewURL(0, 0, stage.stageWidth, stage.stageHeight, "http://www.google.com");

}
```

This does not:

```

var btn1:MySprite = createBtn("openWebView");
btn1.addEventListener(MouseEvent.CLICK, openWebView);
_list.add(btn1);
			
_ex = new RichWebView(stage, true, true, true,true); 
			
function openWebView(e:MouseEvent):void
{
				
				_ex.openWebViewURL(0, 0, stage.stageWidth, stage.stageHeight, "http://www.google.com");

}

```

I'm wonder if for some reason in the iOS debugger, the context for "stage" is lost.

**Comment by myflashlab** [comment]
That makes total sense now. if stage is not available, then the webview size will be zero or NaN. when debugging, you should add a listener to your project to make sure your app is initialized fully before starting any other work on it. Good find.

**Comment by dommermuth** [comment]
Yes, I did add a listener.

this.addEventListener(Event.ADDED_TO_STAGE, onAdded);

Unfortunately, "this.stage" is still lost by the ANE.  I think something
needs to change in your ANE to ensure that the context for stage is
persistent.

**Comment by DigitalSkyline** [comment]
Why is this closed if its not fixed?

**Comment by myflashlab** [comment]
@DigitalSkyline The ```stage``` property is set a bit late on iOS, it seems like a bug on Adobe or Apple side. Anyway, if you are trying to initialize RichWebview as soon as your app runs, you need to wait something around a second or two to make sure the Stage property is set **And also to make sure it has correct width/height dimensions.

**Comment by DigitalSkyline** [comment]
@myflashlab 
Thank you for the reply (oh and btw congrats on the engagement)
... to be clear are you saying after instantiating the webview, wait 1-2 seconds? Because I have verified that the stage size is set prior to the opening call, and I'm doing this on the second frame so I would assume that stage is ready by then (ie, many things happen and are traced out on frame 1 prior to displaying the webview).

`_ex = new RichWebView(stage);
// wait 1-2 seconds???? I'll try this, but doubtful
_ex.openWebViewURL(0, 0, stage.stageWidth, stage.stageHeight, url);`

ex trace log:
`_ex object  main.stage object main.stage.stageWidth 751 main.stage.stageHeight 1334
onPageStarting: http://*****
page started to be loaded [Event type="onPageStarted" bubbles=false cancelable=false eventPhase=2]
page loading: 1%
page loading: 2%
page loading: 3%
page loading: 4%
page loading: 5%
page load completed [Event type="onPageFinished" bubbles=false cancelable=false eventPhase=2]`

and nothing shows up. However, everything works when not in debug mode, and occasionally (about 20% success)  while in debug mode.

Given the closed source, I can't really confirm much more so I'm whole dependant on you for a solution.

Your advice is much appreciated.
